pkg/services/grafana-apiserver/storage/entity/storage.go

```diff
@@ -241,29 +241,16 @@ func (s *Storage) GetList(ctx context.Context, key string, opts storage.ListOpti
 		}
 	}
 
-	// translate grafana.app/folder field selector to folder condition
-	fields := opts.Predicate.Field.Requirements()
-	for _, f := range fields {
-		if f.Field == "grafana.app/folder" {
-			if f.Operator != selection.Equals {
-				return apierrors.NewBadRequest("grafana.app/folder field selector only supports equality")
-			}
-
-			// select items in the spcified folder
-			req.Folder = f.Value
-		}
-	}
-
-	// use Transform function to remove grafana.app/folder field selector
-	opts.Predicate.Field, err = opts.Predicate.Field.Transform(func(field, value string) (string, string, error) {
-		if field == "grafana.app/folder" {
-			return "", "", nil
-		}
-		return field, value, nil
-	})
+	// translate grafana.app/folder field selector to the folder condition
+	fieldRequirements, err := ReadFieldRequirements(opts.Predicate.Field)
 	if err != nil {
 		return err
 	}
+	if fieldRequirements != nil {
+		req.Folder = fieldRequirements.Folder
+	}
+	// Clear the standard fields selector since we are using an explicit version
+	opts.Predicate.Field = nil
```

Comment:
@DanCech -- i think we can just remove the field selector here.  The above clause will either parse supported values (folder) or error, I don't think we need to mix it with the standard field selector process.

Reply:
This is a big hammer, any reason not to just pass opts.Predicate to ReadFieldRequirements and have it remove the field requirements it knows about (folder) or return an updated opts.Predicate?

Removing them altogether here seems like a recipe for confusion later on when we need them for something else, and having to unpick and do the selective removal at that point without the context we have right now.